src/delta.rs

```diff
@@ -152,13 +163,6 @@ impl<'a> StateMachine<'a> {
             .to_string()
         };
         self.line = ansi::strip_ansi_codes(&self.raw_line);
-
-        // Strip the neglected CR.
-        // (CR-LF is unfortunately split by git because it adds ansi escapes between them.
-        //  Thus byte_lines library can't remove the CR properly.)
-        if let Some(b'\r') = self.line.bytes().nth_back(0) {
-            self.line.truncate(self.line.len() - 1);
-        }
     }
```

Comment:
I owe some tests, but we can see that with this commit we are still fixing the `git add -p` pbug that was the reason for the addition of the lines above (now deleted and replaced by the lines further above):

```
git add a.py && git commit -m .
mv b.py a.py
```

Without any \r-stripping, notice that one of the lines is not rendered due to the \r:

<table><tr><td><img width=300px src="https://user-images.githubusercontent.com/52205/143472517-959c0af1-c8e5-4bc3-a8fe-5d8045effb95.png" alt="image" /></td></tr></table> 
With the code after this commit:

<table><tr><td><img width=300px src="https://user-images.githubusercontent.com/52205/143472729-4ad33f53-3965-4729-af04-d502643d21f1.png" alt="image" /></td></tr></table>